src/herder/TransactionQueue.cpp

```diff
@@ -216,21 +145,33 @@ TransactionQueue::shift()
     mBannedTransactions.pop_back();
     mBannedTransactions.emplace_front();
 
+    auto sizes = std::vector<int>{};
```

Comment:
From what I understand: you moved from having a generational `deque` of maps of transactions to a single map so that you can erase all transactions for a given acount if it reaches max age?

I think this is wrong: imagine that there is an account that submits a transaction every second. What is going to happen is that those queues are going to grow but only a subset gets included in each ledger, yet the "age" associated with this account keeps growing, and after 4 ledgers, the queue gets obliterated even though the transactions were getting processed.

Reply:
You are right. We could fix it by resetting all age values for accounts that had their transaction processed in given ledger. I think it makes sense, as we can assume that those accounts have more valid transactions worth processing in their queues and even one bad transaction would clear that queue, so it should be safe.

What do you think?

Reply:
I think this will be good enough, yes: we'll have to make sure that this "reset" only happens when removing transactions that were in a ledger, not when invalid transactions are removed.